iceoryx_posh/include/iceoryx_posh/popo/trigger.hpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_POSH_POPO_TRIGGER_HPP
+#define IOX_POSH_POPO_TRIGGER_HPP
+
+#include "iceoryx_posh/internal/popo/building_blocks/condition_variable_data.hpp"
+#include "iceoryx_posh/popo/trigger_state.hpp"
+#include "iceoryx_utils/cxx/method_callback.hpp"
+
+#include <type_traits>
+#include <typeinfo>
```

Comment:
Is that allowed?

Reply:
@MatthiasKillat surprisingly despite we provide a compile time type to `typeid(T)` it is actually runtime and does not work with RTTI disabled (`-fno-rtti`). For the moment it stays since it improves safety and can be easily removed since it is not required for the actual underlying functionality.